pkgs/tools/audio/openai-whisper-cpp/default.nix

```diff
@@ -28,13 +38,49 @@ stdenv.mkDerivation rec {
   # the models to the current directory of where it is being run from.
   patches = [ ./download-models.patch ];
 
-  nativeBuildInputs = [ makeWrapper ];
+  nativeBuildInputs = [
+      makeWrapper
+    ] ++ lib.optionals cudaSupport ( with cudaPackages ;[
+      cuda_nvcc
+
+      # TODO: Replace with autoAddDriverRunpath
+      # once https://github.com/NixOS/nixpkgs/pull/275241 has been merged
+      autoAddOpenGLRunpathHook
+    ]);
+
+  buildInputs = [
+      SDL2
+    ] ++ lib.optionals stdenv.isDarwin [
+      Accelerate
+      CoreGraphics
+      CoreML
+      CoreVideo
+      MetalKit
+    ] ++ lib.optionals cudaSupport ( with cudaPackages; [
 
-  buildInputs = [ SDL2 ] ++ lib.optionals stdenv.isDarwin [ Accelerate CoreGraphics CoreML CoreVideo MetalKit ];
+      # A temporary hack for reducing the closure size, remove once cudaPackages
```

Comment:
@hughobrien , @amalgame21 Hi, I cannot build this on my system
```
environment.systemPackages = with pkgs; [ (openai-whisper-cpp.override { config = { cudaSupport = true; }; }) ];
```
To fix need to add pkg `cuda_cccl.dev`
Just like here  https://github.com/NixOS/nixpkgs/blob/ecbaaa8de8e25fbcf425cd1a47aca955f6322c01/pkgs/by-name/ll/llama-cpp/package.nix#L67C5-L67C32

```suggestion
      cuda_cccl.dev # <nv/target>
      
      # A temporary hack for reducing the closure size, remove once cudaPackages
```

My failure build log:

```
       > Running phase: buildPhase
       > build flags: SHELL=/nix/store/9vafkkic27k7m4934fpawl6yip3a6k4h-bash-5.2-p21/bin/bash main stream command
       > /nix/store/9vafkkic27k7m4934fpawl6yip3a6k4h-bash-5.2-p21/bin/bash: line 1: which: command not found
       > expr: syntax error: unexpected argument '11.6'
       > I whisper.cpp build info:
       > I UNAME_S:  Linux
       > I UNAME_P:  unknown
       > I UNAME_M:  x86_64
       > I CFLAGS:   -I.              -O3 -DNDEBUG -std=c11   -fPIC -D_XOPEN_SOURCE=600 -D_GNU_SOURCE -pthread -mavx -mavx2 -mfma -mf16c -msse3 -mssse3 -DGGML_USE_CUBLAS -I/usr/local/cuda/include -I/opt/cuda/include -I/targets/x86_64-linux/include
       > I CXXFLAGS: -I. -I./examples -O3 -DNDEBUG -std=c++11 -fPIC -D_XOPEN_SOURCE=600 -D_GNU_SOURCE -pthread -mavx -mavx2 -mfma -mf16c -msse3 -mssse3 -DGGML_USE_CUBLAS -I/usr/local/cuda/include -I/opt/cuda/include -I/targets/x86_64-linux/include
       > I LDFLAGS:  -lcuda -L/nix/store/dkl0bxljckizkn8br44v87hp8dnahqpg-cuda_cudart-12.2.140-lib/lib/stubs -lcublas -lculibos -lcudart -lcublasLt -lpthread -ldl -lrt -L/usr/local/cuda/lib64 -L/opt/cuda/lib64 -L/targets/x86_64-linux/lib
       > I CC:       gcc (GCC) 12.3.0
       > I CXX:      g++ (GCC) 12.3.0
       >
       > nvcc --forward-unknown-to-host-compiler -arch=all -I. -I./examples -O3 -DNDEBUG -std=c++11 -fPIC -D_XOPEN_SOURCE=600 -D_GNU_SOURCE -pthread -mavx -mavx2 -mfma -mf16c -msse3 -mssse3 -DGGML_USE_CUBLAS -I/usr/local/cuda/include -I/opt/cuda/include -I/targets/x86_64-linux/include -Wno-pedantic -c ggml-cuda.cu -o ggml-cuda.o
       > In file included from /nix/store/bx7qv1akfp142m2gl4l562mig39j6rzp-cuda_cudart-12.2.140-dev/include/cuda_fp16.h:4070,
       >                  from /nix/store/jbv65h6822xfdx5nnirj3h38znxi4mrr-libcublas-12.2.5.6-dev/include/cublas_api.h:77,
       >                  from /nix/store/jbv65h6822xfdx5nnirj3h38znxi4mrr-libcublas-12.2.5.6-dev/include/cublas_v2.h:69,
       >                  from ggml-cuda.cu:102:
       > /nix/store/bx7qv1akfp142m2gl4l562mig39j6rzp-cuda_cudart-12.2.140-dev/include/cuda_fp16.hpp:65:10: fatal error: nv/target: No such file or directory
       >    65 | #include <nv/target>
       >       |          ^~~~~~~~~~~
       > compilation terminated.
       > make: *** [Makefile:215: ggml-cuda.o] Error 1
       For full logs, run 'nix log /nix/store/3dd78bpva1hfxnb69yfdqd7rhxd8z6bh-whisper-cpp-1.5.4.drv'.
```

Can you create new PR please with fix please?

Tested locally by using next overlay:
```nix
{ channels, ... }:

final: prev:

{
  openai-whisper-cpp = channels.unstable.openai-whisper-cpp.overrideAttrs (old: {
    buildInputs = (old.buildInputs or [ ]) ++ [
      channels.unstable.cudaPackages.cuda_cccl.dev # <nv/target>
    ];
  });
}
```